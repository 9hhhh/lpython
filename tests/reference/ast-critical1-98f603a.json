{
    "basename": "ast-critical1-98f603a",
    "cmd": "lfortran --show-ast --no-color {infile} -o {outfile}",
    "infile": "tests/critical1.f90",
    "infile_hash": "4fd4203daeba2059609fc7ac7ce16951a4353f77e4a03a142124000c",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast-critical1-98f603a.stdout",
    "stdout_hash": "3134165ecffd8e23bae1bc2c5f816b2ab28194fc19c29c997a7d3201",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}