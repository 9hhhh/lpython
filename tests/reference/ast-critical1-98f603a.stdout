(TranslationUnit [(Program critical1 [] [(ImplicitNone [])] [(Declaration (AttrType TypeInteger [] ()) [] [(joblist [] [(() () CodimensionStar)] () None) (job [] [] () None)]) (Declaration (AttrType TypeInteger [] ()) [] [(a [] [] () None) (b [] [] () None)])] [(If 0 () (== (FuncCallOrArray this_image [] [] []) 1) [(Read 0 [(()) (())] [] [joblist])] []) (SyncAll 0 []) (DoLoop 0 () () () () () [(Critical 0 () [] [(= 0 job (CoarrayRef joblist [] [] [(() 1 () CodimensionExpr)] [])) (= 0 (CoarrayRef joblist [] [] [(() 1 () CodimensionExpr)] []) (- job 1))]) (If 0 () (> job 0) [(= 0 b (+ (** a 2) (** b 2)))] [(Exit 0 ())])]) (SyncAll 0 [])] [])])
